package/cloudshell/cm/customscript/domain/github_script_downloader.py

```diff
@@ -0,0 +1,226 @@
+
+            # 2. filter out junk files from list
+            files_list_in_path = self._remove_junk_files_and_folders_from_files_list(files_list_in_path)
+
+            # 3. get & verify main.ps1/sh/bash exists
+            main_file_info = self.get_main_file(files_list_in_path)
+
+            # 4. download main file
+            main_file_url = self._build_url_to_download_single_file(url_data, main_file_info)
+            self.logger.info('Downloading main file {} from {}'.format(main_file_info['name'], main_file_url))
+            main_file_name, main_file_txt = self._download_single_file(auth, main_file_url)
+            self.logger.info('Done downloading main file')
+
+            # 5. download additional files
+            # todo: download in parallel
+            additional_files = []
+            for file_info in files_list_in_path:
+                # ignore main file
+                if file_info['name'] == main_file_info['name']:
+                    continue
+
+                file_url = self._build_url_to_download_single_file(url_data, file_info)
+                self.logger.info('Downloading single file {} from {}'.format(file_info['name'], file_url))
+                file_name, file_txt = self._download_single_file(auth, file_url)
+                additional_files.append(ScriptFile(name=file_name, text=file_txt))
+                self.logger.info('Done downloading file {}'.format(file_name))
+
+            return ScriptsData(ScriptFile(main_file_name, main_file_txt), additional_files)
+        '''
+
+    def _get_list_of_files_in_path(self, auth, url_data):
+        list_folder_url = self._build_url_to_list_files_in_path(url_data)
+        self.logger.info('Requesting list of file in directory. API request: {}'.format(list_folder_url))
+        response = requests.get(list_folder_url, headers=self.get_auth_header(auth))
+        files_list_in_path = json.loads(response.text)
+        self._validate_response_list_files(response)
+        self.logger.info('Done getting list of files in dir')
+
+        self.cancel_sampler.throw_if_canceled()
+
+        return files_list_in_path
+
+    def _download_single_file(self, auth, url):
+        # download file in URL directly
+        self.logger.info('Downloading file from {}'.format(url))
+        response = requests.get(url, headers=self.get_auth_header(auth))
+        self.logger.info("Response status code from download file {} was {}".format(url,response.status_code))
+        json_response = json.loads(response.text)
+
+        self._validate_response_single_file(response)
+
+        # get file name and validate
+        file_name = self._get_file_name_from_github_response(json_response)
+        self.logger.info('Downloaded file name: {}'.format(file_name))
+
+        # get file content
+        content_bytes = base64.b64decode(json_response['content'])
+        #file_txt = content_bytes.decode('ascii')
+        file_txt = content_bytes
```

Comment:
looks like this can be written in a single line: 
file_txt = base64.b64decode(json_response['content'])

Reply:
done